lib/root_route_constraint.rb

```diff
@@ -0,0 +1,9 @@
+class RootRouteConstraint
+  def initialize
+    @current = Conference.where('end_date >= ?', Date.current).reorder(start_date: :asc)
+  end
+
+  def matches?(*)
+    @current.count == 1 && @current.first.splashpage.public?
```

Comment:
You need to check for ``@current`` and also for ``splashpage``. Your tests fail because ``splashpage`` is nil!